src/pointer-analysis/value_set.cpp

```diff
@@ -1755,9 +1756,23 @@ exprt value_sett::make_member(
     assert(src.operands().size()==1);
     return make_member(src.op0(), component_name, ns);
   }
+  else if(src.id()==ID_byte_extract_little_endian ||
+          src.id()==ID_byte_extract_big_endian)
+  {
+    const typet subtype=struct_union_type.component_type(component_name);
+    const auto &byte_extract_expr = to_byte_extract_expr(src);
+    if(struct_union_type.id()==ID_union &&
+       subtype==byte_extract_expr.op().type() &&
+       byte_extract_expr.offset().is_zero())
+    {
+      // rewrite byte_extract(X, 0).member to X
+      // if the type of X is that of the member
+      return byte_extract_expr.op();
+    }
```

Comment:
Doesn't the simplifier do this? If it doesn't, should it do so? (I note that we do call upon the simplifier anyway, so if the simplifier were taught about it then it should be addressed in here in all cases?)

Reply:
Will look into it!

Reply:
Done.